### ISSUE TITLE
export component's context 

### ISSUE THREAD

**Issue by ailchenkoDynamo** [radix-ui/primitives#1391]
## Feature request
export component's context 

### Overview
I creating theme props for accordion component and I can not get current component's state for my custom Arrow component. 

### Who does this impact? Who is this for?
Advanced

### Additional context
My example and my Arrow component
![image](https://user-images.githubusercontent.com/98591714/167959251-2231d021-28ad-453f-b3ca-b489a33c895e.png)

**Comment by ailchenkoDynamo** [comment]
> Hi @ailchenkoDynamo,
> 
> The accordion trigger has a data attribute (`data-state="open|closed"`) to help with styling. If this isn't enough and you need the state in JavaScript for some reason, you can control your component.

Why create unnecessary states when you can just use the context and easily reuse?

**Comment by benoitgrelard** [comment]
> Why create unnecessary states when you can just use the context and easily reuse?

We are trying to keep the API surface as minimal as possible. Most use cases are covered with the data attributes, or controlling the primitive.

**Comment by QzCurious** [comment]
Hi @benoitgrelard, I also want the context hook on library level. Here's why:

1. If we need the state for animation, it's simpler to write css transition than animation.
2. The picture is cuptured from [dialog example](https://www.radix-ui.com/docs/primitives/components/dialog). I would want to know the open state in `Content`. So I can utilize it with [`<Transition />` from headlessui](https://headlessui.com/react/transition).
![image](https://user-images.githubusercontent.com/38932402/200586338-475395db-8fb3-4145-9086-7e92be2dfad7.png)
3. [tailwindcss](https://tailwindcss.com/docs/transition-property) animationing in/out a component usually done by transition classes. And it's inconvenient to write animation with tailwindcss.

Although the reasons seem tailwindcss-ish. I think it would apply to other styling solutions.

**Comment by andy-hook** [comment]
@QzCurious The open state is already available to you via the controlled props:

https://codesandbox.io/s/tailwind-transition-6pwf41?file=/src/App.js

**Comment by QzCurious** [comment]
Thanks for response. I understand that I can controll open state. And currently I make my own context for open state so `Content` can `useOpenState`. But it just a bit annoying to make the extra code every time I need that state.

**Comment by andy-hook** [comment]
> But it just a bit annoying to make the extra code every time I need that state.

Yeah I can understand your perspective here, as @benoitgrelard mentioned we want to limited the api surface but this can add additional complexity in some cases, especially if you then need to share this controlled value with other components e.g. in an open api design system.

Thanks for your feedback 🙏
